client/api/CfClient.cs

```diff
@@ -35,69 +35,47 @@ public class CfClient
         private SSEListener listener;
-        public StreamReader streamReader { get; set; }
+        private StreamReader streamReader { get; set; }
         //private ServerSentEvent sse;
         private AnalyticsManager analyticsManager;
 
-        public bool isInitialized { get; set; }
-        internal static CfClient instance;
-        private static readonly object padlock = new object();
+        private bool isInitialized { get; set; }
 
-        public static async Task<CfClient> getInstance(string apiKey, Config config)
-        {
-            lock (padlock)
-            {
-                if (instance == null)
-                {
-                    Log.Information("\n\nSTARTING NEW INSTANCE");
-                    instance = new CfClient(apiKey, config);
-
-                }
-            }
-
-            await instance.Authenticate();
-            await instance.init();
-
-            if (instance.apiKey != apiKey)
-            {
-                Log.Error("Client with different ApiKey exist");
-                throw new ApiException("Client with different ApiKey exist", 0, null, null, null);
-            };
-
-            return instance;
-        }
-
-
-
-        public static CfClient getInstance()
-        {
 
-            if (instance == null)
-            {
-                Log.Error("Client not created yet");
-                throw new ApiException("Client not created yet", 0, null, null, null);
-
-            };
+        // Singleton implementation
+        private static readonly Lazy<CfClient> lazy = new Lazy<CfClient>(() => new CfClient()); 
+        public static CfClient Instance { get { return lazy.Value; }  }
 
-            return instance;
-        }
         /// <summary>
-        /// setter for jwtToken
+        /// Initialize the SDK woth default configuration.
         /// </summary>
-        /// <param name="jwt"></param>
-        public void setjwtToken(string jwt)
+        /// <param name="apiKey">SDK API key.</param>
+        /// <exception cref="io.harness.cfsdk.HarnessOpenAPIService.ApiException">If already initialized</exception>
+        /// <returns>async task when initialization is completed</returns>
+        public async Task Initialize(string apiKey)
         {
-            jwtToken = jwt;
+            await Initialize(apiKey, Config.Builder().Build() );
         }
 
-        public CfClient(string apiKey) : this(apiKey, Config.Builder().Build()) { }
 
-        public CfClient(string apiKey, Config config)
+        /// <summary>
+        /// Initialize the SDK.
+        /// </summary>
+        /// <param name="apiKey">SDK API key.</param>
+        /// <param name="config">SDK configuration.</param>
+        /// <exception cref="io.harness.cfsdk.HarnessOpenAPIService.ApiException">If already initialized</exception>
+        /// <returns>async task when initialization is completed</returns>
+        public async Task Initialize(string apiKey, Config config)
         {
```

Comment:
API initialization is now a public method. Client can use cfClient object in form of singleton, or just instantiate with public constructor, and then initialize.